src/base/memory_monitor_sr.cpp

```diff
@@ -358,25 +446,82 @@ namespace cubperf
     return 0;
   }
 
-  int memory_monitor::add_stat (THREAD_ENTRY *thread_p, MMON_STAT_ID stat_id, uint64_t size, bool expand)
+  void memory_monitor::add_stat (THREAD_ENTRY *thread_p, MMON_STAT_ID stat_id, uint64_t size)
   {
-    return 0;
+    LOG_TDES *tdes;
+    TRANID trid = NULL_TRANID;
+    int module_idx = MMON_GET_MODULE_IDX_FROM_STAT_ID (stat_id);
```

Comment:
1. avoid using macro and replace it to inline function
2. MMON_GET_MODULE_IDX_FROM_STAT_ID
    - int memory_monitor::get_module_index (MMON_STAT_ID stat_id)
3. idx_map_val, comp_idx, subcomp_idx are not needed in here
    - so, change the mmon_module::add_stat() signiture
        - from (uint64_t size, int comp_idx, int subcomp_idx, bool allocation_at_init)
        - to (MMON_STAT_ID stat_id, uint64_t size, bool allocation_at_init)
    - and get the necessary index in there
 4. remove allocation_at_init mechanishm (I aready mentioned it in the previous PR)
    - introduce a new API function to notify(inform) end of the initialization state to memory_monitor
    - for example
        ```
        int mmon_notify_end_of_boot(void)
        {
            // interate modules and componts
            // copy cur_stat to init_stat
            // etc. something you need to handle
        }
        ```

Reply:
5. based on the definition of the init_stat as I know, the LOG_ISRESTARTED() checking is not enough; because the log_Gl.rcv_phase is set to LOG_RESTARTED in the log_initialize() function stack
    - BOOT_SERVER_UP is the better option for you

Reply:
Then `BO_IS_SERVER_RESTARTED ()` can be used instead

Reply:
Done.